ouroboros-network/src/Ouroboros/Network/Subscription/Common.hs

```diff
@@ -143,40 +162,18 @@ subscribeTo tbl threadPool tracer localIPv4_m localIPv6_m connectionAttemptDelay
                     activeCons  <- readTVar conThreads
                     unless (null activeCons) retry
 
-                valencyLeft <- atomically $ readTVar valencyVar
+                valencyLeft <- atomically $ readValencyCounter valencyVar
                 if valencyLeft == 0
                    then traceWith tracer SubscriptionTraceSubscriptionRunning
                    else traceWith tracer SubscriptionTraceSubscriptionFailed
 
-
-    doConnect conThreads localAddr remoteAddr =
-        bracket
-            ( do
-                tid <- myThreadId
-                atomically $ do
-                    modifyTVar' conThreads (Set.insert tid)
-                    modifyTVar' threadPool (Set.insert tid)
-                sd <- Socket.socket (sockAddrFamily remoteAddr) Socket.Stream
-                        Socket.defaultProtocol
-                hasRefVar <- newTVarM False
-                return (sd, hasRefVar)
-            )
-            (\(sd,hasRefVar) -> do
-                traceWith tracer $ SubscriptionTraceConnectCleanup remoteAddr
-
-                tid <- myThreadId
-                atomically $ do
-                    modifyTVar' conThreads (Set.delete tid)
-                    modifyTVar' threadPool (Set.delete tid)
-                hasRef <- atomically $ readTVar hasRefVar
-                when hasRef $ do
-                    localAddr' <- Socket.getSocketName sd
-                    removeConnection tbl remoteAddr localAddr'
-
-                Socket.close sd
-
-            )
-            (\(sd,hasRefVar)  -> do
+    doConnect
+        :: TVar IO (Set ThreadId)
+        -> Socket.SockAddr
+        -> Socket.SockAddr
+        -> (Socket.Socket, TVar IO Bool)
+        -> IO ()
+    doConnect conThreads localAddr remoteAddr (sd,hasRefVar) = do
                 Socket.setSocketOption sd Socket.ReuseAddr 1
```

Comment:
I know it's outside of this PR, but the indentation is very broad here.

Reply:
It is very much related to this PR, as I caused it while addressing other review comments :)
Will fix.